integration/chartconfig/chartconfig.go

```diff
@@ -0,0 +1,82 @@
+package chartconfig
+
+import (
+	"bytes"
+	"fmt"
+	"html/template"
+
+	"k8s.io/helm/pkg/helm"
+
+	"github.com/giantswarm/apprclient"
+	"github.com/giantswarm/e2etemplates/pkg/e2etemplates"
+	"github.com/giantswarm/helmclient"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/spf13/afero"
+)
+
+type ChartConfigValues struct {
+	Channel              string
+	ConfigMap            ConfigMap
+	Name                 string
+	Namespace            string
+	Release              string
+	Secret               Secret
+	VersionBundleVersion string
+}
+
+type ConfigMap struct {
+	Name            string
+	Namespace       string
+	ResourceVersion string
+}
+
+type Secret struct {
+	Name            string
+	Namespace       string
+	ResourceVersion string
+}
+
+func (ccv ChartConfigValues) ExecuteChartValuesTemplate() (string, error) {
+	buf := &bytes.Buffer{}
+	chartValuesTemplate := template.Must(template.New("chartConfigChartValues").Parse(e2etemplates.ApiextensionsChartConfigE2EChartValues))
+	err := chartValuesTemplate.Execute(buf, ccv)
+	if err != nil {
+		return "", err
+	}
+
+	return buf.String(), nil
+}
+
+func (ccv ChartConfigValues) UpdateChartOperatorResource(logger micrologger.Logger, helmClient *helmclient.Client, releaseName string) error {
```

Comment:
This will be improved in future PRs, I think the logger should be part of the struct that we operator on here. So I might create a `ChartConfig` struct which holds the `ChartConfigValues` and a `Logger`.

Reply:
I was wondering if we want to have some general primitive for updating chart values within our tests. Maybe some food for thought for the future.

Reply:
IMO this is mixing pure data structures with structures providing logic via their methods. In general I think `ChartConfigValues` should have any method performing business logic, i.e. other than `New`, `GoString`, `String`, `MarshalToJson`, etc.

Reply:
I would prefer having separate struct holding data (as `ChartConfigValues`) and separate one performing operations using `ChartConfigValues` as an argument.